doc/debugger.rst

```diff
@@ -1773,6 +1809,105 @@ target endianness and interpret the bytecode based on that.
 .. note:: This command is somewhat incomplete at the moment and may be modified
    once the best way to do this in the debugger UI has been figured out.
 
+AppRequest request (0x23)
+-------------------------
+
+Format::
+
+    REQ <int: 0x23> [<tval>*] EOM
+    REP [<tval>*] EOM
+
+Example::
+
+    REQ 35 "GameInfo" "GetTitle" EOM
+    REP "Spectacles: Bruce's Story" EOM
+
+This is a custom request message whose meaning and semantics depend on the
+application.
+
+AppRequest messages are used for direct communication between the debug client
+and debug target over the Duktape debug protocol.  Both the meaning of a custom
+message and the dvalues it contains are entirely up to the implementation and
+depending on the needs of the application, need not be supported at all.
+
+See "Custom requests and notifications" below for more details.
+
+Custom requests and notifications
+=================================
+
+Starting in Duktape 1.5.x, Duktape supports direct communication between the
+debug client and debug target over the same transport by using the special
+'AppRequest' and 'AppNotify' messages. These messages have no meaning to
+Duktape, which merely serves to marshal them back and forth through a defined
+API.
+
+AppNotify messages may be sent by pushing the contents of the message to the
+stack and calling `duk_debugger_notify()` passing the number of values pushed.
+Each value pushed will be sent as a dvalue in the message. So if you push two
+strings, "foo" and "bar", the client will see ``NFY 7 "foo" "bar" EOM``.
+
+AppRequests are used to make requests to the target which are not directly
+related to Ecmascript execution. For example, an AppRequest might be used to:
+
+* Change the frame rate of a game engine
+* Reset/reboot an embedded target device over the debug protocol
+* Perform or trigger software or script updates
+
+Supporting this requires attaching using the ``duk_debugger_attach_custom()``
```

Comment:
"Supporting this" seems a bit vague in this context, maybe explicitly "Supporting application requests requires..." or "Supporting AppRequest requires..."